docs/arcaflow/workflows/flow-control.md

```diff
@@ -2,6 +2,289 @@
+`disabled`. The enabled output is the expression provided, and the disabled output is automatically generated from
+the expression provided.
+
+The input must be in one of the following formats:
+- `$.steps.step_name.outputs`
+- `steps.step_name.outputs`
+- `$.steps.step_name.outputs.output_name`
+- `steps.step_name.outputs.output_name`
+- `$.steps.step_name.outputs.output_name.output_field`
+- `steps.step_name.outputs.output_name.output_field`
+
+Using `!ordisabled`, the following example is otherwise equivalent to the prior example:
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !ordisabled $.steps.my_step.outputs.success
+```
+
+#### Alternative methods
+
+For handling disabled steps, `!oneof` and `!ordisabled` are the recommended methods because they cause output failure
+when the step fails. However, if it is acceptable for the workflow to succeed when a step crashes, the optional tags
+could be the right feature for your workflow.
+
+The alterative methods are the "optional" family of expression tags. The `oneof` tags instructed the workflow to
+build a `oneof_string` type with OR dependencies, requiring one of the options to have an output for the oneof to
+resolve. Meanwhile, the "optional" family of tags can resolve without an output by utilizing optional object property
+fields.
+
+| Tag              | Description                                      |
+|------------------|--------------------------------------------------|
+| `!wait-optional` | The typical one that should be used. It will resolve either when the value becomes present, or when it is known that the value will not resolve. |
+| `!soft-optional` | Not recommended for most use cases. It will not wait for the value to be ready. If the value is present, it will set the field. If it isn't, it will exclude it. |
+
+##### How to use `!wait-optional`
+
+```yaml title="workflow.yaml"
+version: v0.2.0
+input:
+  root: RootObject
+  objects:
+    RootObject:
+      id: RootObject
+      properties:
+        step_1_enabled:
+          type:
+            type_id: bool
+        step_1_input:
+          type:
+            type_id: integer
+steps:
+  my_step:
+    plugin:
+      deployment_type: image
+      src: path/to/my_plugin:1
+    input:
+      param_1: !expr $.input.step_1_input
+    enabled: !expr $.input.step_1_enabled
+outputs:
+  workflow_success:
+    plugin_output: !wait-optional $.steps.my_step.outputs.success
```

Comment:
How is the behavior of the workflow affected by the change from `!expr` to the `!wait-optional`? It seems like nothing is different for a standalone workflow.

Reply:
With !expr, there is an AND dependency, causing failure if the plugin is disabled or crashes. With `!wait-optional`, there is a completion dependency, so it will not fail in the event of the step failing. It would resolve as an empty field if it is not present.

Do you want this clarified in any way in the documentation?

Reply:
No, I don't think it needs to be added